src/main/java/com/cburch/logisim/fpga/data/IOComponentTypes.java

```diff
@@ -270,12 +398,32 @@ public static void paintPartialMap(Graphics2D g,
     var hasDp = false;
     switch (type) {
       case DIPSwitch: {
-        final var part = (width > height) ? (float) width / (float) nrOfPins : (float) height / (float) nrOfPins;
-        final var bx = (width > height) ? x + (int) ((float) pinNr * part) : x;
-        final var by = (width > height) ? y : y + (int) ((float) pinNr * part);
-        final var bw = (width > height) ? (int) ((float) (pinNr + 1) * part) - (int) ((float) pinNr * part) : width;
-        final var bh = (width > height) ? height : (int) ((float) (pinNr + 1) * part) - (int) ((float) pinNr * part);
-        g.fillRect(bx, by, bw, bh);
+        var part = (float) 0;
+        var boxXpos = 0;
+        var boxYpos = 0;
+        var boxWidth = 0;
+        var boxHeight = 0;
+        var yPinNr = pinNr;
+        switch (mapRotation) {
+          case rotationPlusNinety : yPinNr = nrOfPins - pinNr - 1;
+          case rotationMinusNinety : {
+            part = (float) height / (float) nrOfPins;
+            boxXpos = x;
```

Comment:
I'd group `boxXpos` and `boxYpos`. Also I'd rename it to i.e. `boxPosX` and `boxPosY` which looks better and still satisfy bean naming convention.